worktree.go

```diff
@@ -188,6 +189,67 @@ func (w *Worktree) Checkout(opts *CheckoutOptions) error {
 
 	return w.Reset(ro)
 }
+
+// CreateOrphanBranch creates a new orphan branch with an initial commit with msg as the commit message
+func (w *Worktree) CreateOrphanBranch(branch plumbing.ReferenceName, msg string, co *CommitOptions) error {
```

Comment:
@john-cai when I use this it seems to keep the worktree from the current branch, I'm looking for something that essentially replicates: `git switch --orphan my-branch` but this implementation replicates `git checkout --orphan my-branch` instead (a flag maybe? `keepWorkingTree`). 

I'm also wondering if you could allow creating an orphan branch without committing to it, so we can add whatever files ourselves before making a commit on the orphan branch.

Reply:
i've been hacking around with this and it seems if we clear the index and then construct the tree using that index instead of taking the tree from head then we can make the initial commit empty, e.g. :
```golang
	wt, err := w.r.Worktree()

	if err := w.RemoveGlob("*"); err != nil {
		return err
	}

	idx, err := w.r.Storer.Index()
	if err != nil {
		return err
	}

	h := &buildTreeHelper{
		fs: w.Filesystem,
		s:  w.r.Storer,
	}

	tree, err := h.BuildTree(idx)
	if err != nil {
		return err
	}
```

and then pass `tree` to `buildCommitObject`

Reply:
which sounds like what `switch --orphan` does according to the help text:
```
Create a new orphan branch, named <new-branch>. All tracked files are removed.
```

Reply:
maybe instead we can have a flag to say whether the tree is built from index or head, that way users can get the index in whatever state they want before the initial commit is made

Reply:
I've been playing around with this more today and realised that we don't actually need any of this to create an orphaned branch.

`repo.Storer.SetReference(plumbing.NewSymbolicReference(plumbing.HEAD, branch))` is all you need, and it won't create any commits so is closer to the `git checkout --orphan` behaviour. To replicate the `git switch --orphan` before you can just do `worktree.RemoveAll("*")` after.